### DISCUSSION TITLE
Raspberry pi 5

### DISCUSSION THREAD

**Discussion by pkombala** [probonopd/MiniDexed#635] [Q&A] [answered]
Sir, please help me if any update it was not booting

**Comment by Kirtai** [comment]
Circle (the code for bare hardware programming on the Pi) isn't ready for the Pi5 yet.

**Reply by probonopd** [reply]
Relevant discussion:

https://github.com/rsta2/circle/discussions/413

But at this point using a RPi 5 for MiniDexed would be overkill anyways - unless we somehow make use of all that additional power...

**Reply by Kirtai** [reply]
True. If the Pi4 can handle double polyphony, improved effects *and* maybe even more tone generators, what is there to add that would need all that power?

**Comment by diyelectromusic** [comment]
For anyone feeling brave (and would like to let me know what isn't working) :)

This is NOT a supported configuration yet mind - you need to be able to build it yourself!

https://github.com/probonopd/MiniDexed/pull/638

Kevin

**Reply by probonopd** [reply]
Very cool, Kevin!
There is now a build for testing linked in https://github.com/probonopd/MiniDexed/pull/638.

**Comment by diyelectromusic** [comment, marked as answer]
This has now been merged into the main release.  Please give it a go and feel free to report back here any issues or thoughts.

**_Note that as far as I'm concerned, this is still experimental and I recommend caution building your first MiniDexed with a more expensive Pi 5!  Also, I don't know what will happen cooling wise either._**

Note: In terms of performance, I've not really seen much need personally to go beyond a Pi 3 :)

But it is nice to be able to use MiniDexed on any Pi.  At some point we might experiment with more polyphony and tone generators (see discussion here if you want to follow that for the Pi 4 and 5: https://github.com/probonopd/MiniDexed/pull/623)

Kevin

**Reply by temporal-distortion** [reply]
I'm terribly sorry that i'm late to this discussion. I just recently began my journey into Minidexed (my ambition is to build it into a proper 19-inch rack with 20x2 LCD, audio-hat, midi-hat, external connectors for RCA/Tele, midi, power and externally accessible micro-sd slot.

While i have commented elsewhere here about my issues with rPI4 (https://github.com/probonopd/MiniDexed/discussions/929) i have had great success with both rPI3b+ and rPI5.

I'm not completely sure but i think the rPI4 and above can provide 32 note polyphony (?) and so of course for my the rPI5 would be the best option. When running minidexed (and others such as the jv880) on the 5, i haven't really noticed any issues at all. The fan is very quiet (maybe because its not even struggling with the load).

When reading about using a fanless rPI5, the comments usually states that it might not be ideal (throttling) but that it will still perform better than a rPI4 anyway. So with this in mind, i guess its not such a bad thing to actually use a rPI5 for Minidexed and perhaps the day will come when PWM controlled cpu fan will be supported (?).

Just me thinking aloud :)

**Reply by Kirtai** [reply]
Maybe you could use a large passive heat sink attached to the cast instead of a fan.
I believe the Zynthian vendor sells a custom heatblock specifically for this purpose.

**Reply by diyelectromusic** [reply]
Pi 4 or 5 can support up to 32 note polyphony, but we drop it to 24 by default on the Pi 4. I think that was just to be cautious as some of the performances might start to stress it and cause audio artefacts when on full 32 note polyphony on a Pi 4.  But I also seem to recall most were fine, so see how you go? (assuming you get it working!!) :)

See: https://github.com/probonopd/MiniDexed/blob/main/src/config.h#L70

You'll probably be fine with a Pi 5, but it is still early days really.  We just didn't want to be responsible for someone trashing their more expensive board when the cheaper ones have a lot more "in the field" use so far, so it is definitely "at your own risk".  But someone has to be first I guess!

Kevin

**Reply by Banana71** [reply]
> Pi 4 or 5 can support up to 32 note polyphony, but we drop it to 24 by default on the Pi 4. I think that was just to be cautious as some of the performances might start to stress it and cause audio artefacts when on full 32 note polyphony on a Pi 4. But I also seem to recall most were fine, so see how you go?

I've been running my Raspberry Pi 4 without a heatsink and, ever since it became possible, always with 32-voice polyphony. I'm not aware of any performance issues. @diyelectromusic Kevin, can you tell me more about the problems?

Peter

**Reply by diyelectromusic** [reply]
No sorry, can't really remember - I think I was just being cautious when upgrading the code from 16 to make sure we had a known-good default :)

Kevin

**Reply by Banana71** [reply]
Or was it dependent on the engine type? Mark I and OPL are a bit more demanding when using algorithms 4 or 6.

That's why I only use the Modern engine type, even though the two algorithms with the large feedback don't work.
